packages/atomic/src/utils/store.ts

```diff
@@ -1,26 +1,42 @@
 import {NumericFacetValue, DateFacetValue} from '@coveo/headless';
+import {ObservableMap} from '@stencil/store';
 
-interface FacetLabel {
+interface FacetInfo {
   label: string;
 }
 
 interface FacetValueFormat<ValueType> {
   format(facetValue: ValueType): string;
 }
 
-type FacetStore<F extends FacetLabel> = Record<string, F>;
+type FacetType = 'facets' | 'numericFacets' | 'dateFacets' | 'categoryFacets';
+type FacetStore<F extends FacetInfo> = Record<string, F>;
 
 export type AtomicStore = {
-  facets: FacetStore<FacetLabel>;
-  numericFacets: FacetStore<FacetLabel & FacetValueFormat<NumericFacetValue>>;
-  dateFacets: FacetStore<FacetLabel & FacetValueFormat<DateFacetValue>>;
-  categoryFacets: FacetStore<FacetLabel>;
+  facets: FacetStore<FacetInfo>;
+  numericFacets: FacetStore<FacetInfo & FacetValueFormat<NumericFacetValue>>;
+  dateFacets: FacetStore<FacetInfo & FacetValueFormat<DateFacetValue>>;
+  categoryFacets: FacetStore<FacetInfo>;
+  facetElements: HTMLElement[];
 };
 
 export const initialStore: () => AtomicStore = () => ({
   facets: {},
   numericFacets: {},
   dateFacets: {},
   categoryFacets: {},
-  refineEnabled: false,
+  facetElements: [],
 });
+
+export const registerFacetToStore = <T extends FacetType, U extends string>(
+  store: ObservableMap<AtomicStore>,
+  facetType: T,
+  data: AtomicStore[T][U] & {facetId: U; element: HTMLElement}
+) => {
+  if (store.state[facetType][data.facetId]) {
+    return;
+  }
+
+  store.state[facetType][data.facetId] = data;
+  store.state.facetElements.push(data.element);
```

Comment:
It feels like this change is creating a second source of truth for where to find facet elements. If the ids ever change, then the `facetElements` array needs to be updated.

Could we instead use a function to dynamically return the elements? e.g.

```
function getFacetElements(store) {
    // return facets, date facets, numeric facets, category facets
}
```

Reply:
We don't want the same facet id to be registered twice, that is done at line 36. 

We also have the numeric and date filter which are registered in the store under numeric & date facets, they have separate facet ids, but we don't want to see them as separate elements, so both dataset (the labels and formats VS the elements) are a bit different.

If the facetId changes, you would have to call this method again to update the labels anyways.

Reply:
I see, thanks for the clarifications,